pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -366,7 +366,10 @@ func (gc *GarbageCollector) attemptToDeleteWorker(ctx context.Context, item inte
 		// a cluster-scoped object referring to a namespaced owner is an error that will not resolve on retry, no need to requeue this node
 		return forgetItem
 	} else if err != nil {
-		if _, ok := err.(*restMappingError); ok {
+		if errors.IsConflict(err) {
```

Comment:
Summarizing the slack conversation.  The most preferred option is to have a precondition on ownerReferences remaining the same, but that's at least a release or two out.  Lacking that, having a resourceVersion precondition and on conflicts checking to see if the new live object (we always do a live get before delete) has the same ownerReferences, we issue a delete without the resourceVersion precondition. This prevents hot-looping writes of fields other than ownerReferences from blocking GC.